lib/px/service/client/base.rb

```diff
@@ -4,6 +4,19 @@ class Base
     include Px::Service::Client::CircuitBreaker
     cattr_accessor :logger
 
+    def initialize
+      # monitors circuit state and throws CircuitBrokenException
+      # if state is open. Similar to `handle` function of circuit_handler
+      handler = self.class.circuit_handler
+      Typhoeus.before do
+        if handler.is_tripped(self.circuit_state)
```

Comment:
good point. we could clear out the queue after the first `before` call if circuit breaker is in half_open state. allowing only 1 request to go through. worst case scenario is we lose max-concurrent request -1 successful requests. best case scenario is we stop same number of fail requests